### PR TITLE
Use method to set model to take advantage of Promises (closes #37)

### LINKED ISSUE #37: Should model be set via method instead of setter?
If model were set via a function then we could return a `Promise` that could indicate when the task was complete. This would negate the need for more events or special named Promises on the instance (like `.complete` or `.rendered`).

``` js
list.model(myModel).then(...)
```

The Promise would resolve when the list render was fully complete and interactive. If the user wanted to know when the first render has completed they could use the specific `.rendered` Promise, that would be a one time thing.

``` js
list.model(myModel).then(() => {
  // setup complete and interactive
});

list.rendered.then(() => {
  // ready to show
})
```

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

File: gaia-fast-list.js [hunk 1]
```diff
@@ -894,6 +913,7 @@ Internal.prototype = {
   cacheImage(image, index) {
     if (!this.el.imageCacheLength) return;
     if (!this.el.imageCacheSize) return;
+    if (this.images.hash[index]) return;
```
Comment (wilsonpage) [inline gaia-fast-list.js:903]: This guards against us very rarely attempting to cache the same image twice. Was causing intermittent test failures.

File: test/test.js [hunk 2]
```diff
@@ -28,29 +28,25 @@ suite('GaiaFastList >>', function() {
 
   test('it creates FastList when model is first set', function() {
     var el = createList();
```
Comment (wilsonpage) [inline test/test.js:29]: Having these Promises around has removed a lot of code from our test-suite :)

**Comment by wilsonpage** [comment]
@etiennesegonzac r?

File: gaia-fast-list.js (hunk 1 above)
Comment (etiennesegonzac) [inline gaia-fast-list.js:903]: can you rebase this one on master to trigger a new build and see if the intermittent goes away?

**Comment by etiennesegonzac** [comment]
But looks really good!

**Comment by wilsonpage** [comment]
Changed `.model()` to `.setModel()`.

**Comment by wilsonpage** [comment]
Ready to land ... :airplane:

**Pull request merged**